participant/e2e/utils/routing.spec.ts

```diff
@@ -0,0 +1,150 @@
+// routing tests
+import { test, expect } from "@playwright/test";
+import {
+  fillParticipantForm,
+  fillCountForm,
+  fillChangesForm,
+  fillNameForm,
+  fillUploadForm,
+  fillContactForm,
+} from "../helpers/formFillers";
+import type { Participant } from "~/types";
+import { readFileSync } from "fs";
+
+const participantData: Participant = {
+  dob: { day: 3, year: 2004, month: 2 },
+  tag: "TtmTDA5JcBAWr0tUWWmit",
+  firstName: "Delightful",
+  lastName: "Cheesemuffin",
+  adjunctive: "no",
+  relationship: "child",
+};
+
+for (const routeOK of ["/", "/about", "/name"]) {
+  test(`without forms filled, can request ${routeOK}`, async ({ page }) => {
+    await page.goto(`/missoula/recertify${routeOK}`, {
+      waitUntil: "networkidle",
+    });
+    await expect(page).toHaveURL(`/missoula/recertify${routeOK}`);
+  });
+}
+
+test(`with name form filled, can request expected routes`, async ({ page }) => {
+  await fillNameForm(page, "Matt", "Gardener", "/gallatin/recertify/count");
+  for (const routeOK of ["/", "/about", "/name", "/count", "/details"]) {
+    await page.goto(`/missoula/recertify${routeOK}`, {
+      waitUntil: "networkidle",
+    });
+    await expect(page).toHaveURL(`/missoula/recertify${routeOK}`);
+  }
+});
+
+test(`with name and details forms filled, can request expected routes`, async ({
+  page,
+}) => {
+  await fillNameForm(page, "Matt", "Gardener", "/gallatin/recertify/count");
+  await fillCountForm(page, 1, "/gallatin/recertify/details?count=1");
+  await fillParticipantForm(page, participantData, 0);
+  await page.getByRole("button", { name: "Continue" }).click();
+  await expect(page).toHaveURL("/gallatin/recertify/changes");
+  for (const routeOK of ["/", "/about", "/name", "/details", "/changes"]) {
```

Comment:
Is this technically also ok?
```suggestion
  for (const routeOK of ["/", "/about", "/name", "/count", "/details", "/changes"]) {
```

Reply:
And same with the subsequent tests?

Reply:
No, because once the participant form is done, you can't route to "/count" anymore
line #48 submits the details form
`await page.getByRole("button", { name: "Continue" }).click();`

Reply:
(and, subsequent tests that have participant details filled can't navigate to count either)

Reply:
Got it. I missed that in `routing.ts`. It's a very good solution to preventing that weird count behavior after someone has filled in participant data.